### ISSUE TITLE
How to avoid the trivial solution?

### ISSUE THREAD

**Issue by iamzyj** [lululxvi/deepxde#321]
Hello! I'm trying to solve a 1D, steady-state Schrödinger equation, and I started at a very simple situation:
0.5*y''+y=0, with boundary condition:

- y(0)=1

- y'(0)=0

The exact solution is y(x)=cos(sqrt(2)*x), while what I've got is something like a δ function. It makes sense because it's a trivial solution.

<img width="617" alt="微信截图_20210627174115" src="https://user-images.githubusercontent.com/83071043/123550772-6f163480-d76f-11eb-86b5-2b10c433bdd6.png">

```
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
import numpy as np
import deepxde as dde
from deepxde.backend import tf
from tensorflow.python.ops.gen_math_ops import sqrt

def main():
    def ddy(x, y):
        return dde.grad.hessian(y, x)

    def pde(x, y):

        return y+0.5*ddy(x, y)  


    def boundary_m(x, _):
        return np.isclose(x[0], 0)



    geom = dde.geometry.Interval(-300, 300)

    bc1 = dde.DirichletBC(geom, lambda x: 1, boundary_m)
    bc2 = dde.NeumannBC(geom, lambda x: 0, boundary_m)
   

    data = dde.data.PDE(
        geom,
        pde,
        [bc1, bc2],
        num_domain=1000,
        num_boundary=200,
        anchors=np.array([0.0])
    )
    layer_size = [1] + [50] * 3 + [1]
    activation = "tanh"
    initializer = "Glorot uniform"
    net = dde.maps.FNN(layer_size, activation, initializer)

    model = dde.Model(data, net)
    model.compile("adam", lr=0.001)
    losshistory, train_state = model.train(epochs=10000)

    dde.saveplot(losshistory, train_state, issave=True, isplot=True)

if __name__ == "__main__":
    main()
```

Thanks!

**Comment by lululxvi** [comment]
Could you first try a small domain, e.g., [-1, 1]? Also, you may try to use larger weight for the BC losses, see FAQ on how to do this?

**Comment by iamzyj** [comment]
> Could you first try a small domain, e.g., [-1, 1]? Also, you may try to use larger weight for the BC losses, see FAQ on how to do this?
Thanks for your reply!
Yes, it works out fine with a small domain. But for a larger domain, changing loss weight has little impact on it. Maybe I can iterate for a bunch of small domains.
And I have another question, for those who have multiple solutions, besides specifying BCs, is there any other way to avoid falling into the trival solution?

**Comment by lululxvi** [comment]
The trival solution is due to the network optimization. Usually, some hyperparameter tuning would help. There are also other problem-dependent possibilities.

**Comment by iamzyj** [comment]
> The trival solution is due to the network optimization. Usually, some hyperparameter tuning would help. There are also other problem-dependent possibilities.
> 
> For this problem, you may also try hard constraints for the IC/BC, see FAQ.

Thanks! I've applied hard constraints on an equation with potential energy, and it worked as expected.
